src/wzapi.cpp

```diff
@@ -2356,7 +2377,18 @@ bool wzapi::gameOverMessage(WZAPI_PARAMS(bool gameWon, optional<bool> _showBackD
 		}
 	}
 //	jsDebugMessageUpdate();
-	displayGameOver(gameWon, showBackDrop);
+	if (bMultiPlayer && NetPlay.players[selectedPlayer].isSpectator)
+	{
+		// Display a custom game over screen for spectators (win or loss conditions don't apply, obviously)
+		clearMissionWidgets();
+		intAddMultiMenu();
+		addConsoleMessage(_("GAME OVER"), CENTRE_JUSTIFY, SYSTEM_MESSAGE, false, MAX_CONSOLE_MESSAGE_DURATION);
+		addConsoleMessage(_("The battle is over - you can leave the room."), CENTRE_JUSTIFY, SYSTEM_MESSAGE, false, MAX_CONSOLE_MESSAGE_DURATION);
```

Comment:
Bad wording.
In ffa mode, you see this message too early.
If you are a host observer too.